### DISCUSSION TITLE
Differences using "~"

### DISCUSSION THREAD

**Discussion by JTorresConsulta** [automadcms/discussions#5] [General]
What the differences using ~ in the v1 templates code?

I can`t find an answer in the official docs:
https://automad.org/developer-guide/cheat-sheets/useful-template-snippets

```
<@ if @> <@ end @>

<@~ if@> <@~ end @>

<@~ if ~@> <@~ end ~@>
```

Thanks and congratulations for this cool CMS (v1)

**Comment by marcantondahmen** [comment]
Hi, thank you! The `~` is used to trim whitespace before or after a statement.

```
Text 

<@~ ... @> 
```

This will trim all whitespace to the left and is essentially the same as

```
Text<@ ... @> 
```

Accordingly 

```
<@ ... ~@> 
```

will remove all whitespace to the right.

This comes in handy when whitespace is not wanted in the rendered output but the template itself should be readable.
